<?xml version="1.0"?>
<launch>
    <arg name="markers_file_name" default="map_markers_1"/>
    <arg name="marker_positions_file" default="$(find environments)/maps/kitchen_dining/$(arg markers_file_name).yaml" />
    <arg name="cameras_file_name" default="cameras"/>
    <arg name="cameras_positions_file" default="$(find amcl_miguel)/maps/$(arg cameras_file_name).yaml" />
    <arg name="scan_topic" default="camera/scan/noise"/>
    <arg name="odom_topic" default="odom"/>
    <arg name="base_frame_robot" default="base_footprint"/>
    <arg name="global_frame" default="map"/>
    <arg name="camera_link" default="camera_link"/>
    <arg name="camera_topic" default="/camera/RGB/camera_info"/>
    <arg name="topic_output_video" default="/image_converter/output_video"/>
    <arg name="topic_marker" default="/detected_markers"/>   
    <arg name="initial_pose_x" default="-1.631"/>
    <arg name="initial_pose_y" default="-3.923"/>
    <arg name="initial_pose_a" default="1.57"/>
    <arg name="height_pos_camera_link" default="-0.0001"/> <!-- If you put value -0.0001, this variable will be calculated using tf topic -->


    <node pkg="amcl_miguel" type="amcl_miguel" name="amcl_miguel" respawn="false" output="screen" >
    
        <param name="initial_pose_x"    value="$(arg initial_pose_x)"/>
        <param name="initial_pose_y"    value="$(arg initial_pose_y)"/>
        <param name="initial_pose_a"    value="$(arg initial_pose_a)"/>

        <!-- Publish scans from best pose at a max of 10 Hz -->
        <param name="odom_model_type" value="diff"/>
        <param name="odom_alpha5" value="0.1"/>
        <param name="transform_tolerance" value="0.2" />
        <param name="gui_publish_rate" value="10.0"/>
        <param name="laser_max_beams" value="30"/>
        <param name="min_particles" value="500"/>
        <param name="max_particles" value="5000"/>
        <param name="kld_err" value="0.05"/>
        <param name="kld_z" value="0.99"/>
        <param name="odom_alpha1" value="0.2"/>
        <param name="odom_alpha2" value="0.2"/>
        <!-- translation std dev, m -->
        <remap from="scan" to="$(arg scan_topic)"/>
        <param name="odom_alpha3" value="0.8"/>
        <param name="odom_alpha4" value="0.2"/>
        <param name="laser_z_hit" value="0.5"/>
        <param name="laser_z_short" value="0.05"/>
        <param name="laser_z_max" value="0.05"/>
        <param name="laser_z_rand" value="0.5"/>
        <param name="laser_sigma_hit" value="0.1"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_model_type" value="likelihood_field"/>
        <param name="laser_likelihood_max_dist" value="2.0"/>
        <param name="update_min_d" value="0.2"/>
        <param name="update_min_a" value="0.5"/>
        <param name="resample_interval" value="1"/>
        <param name="transform_tolerance" value="0.1"/>
        <param name="recovery_alpha_slow" value="0.0"/>
        <param name="recovery_alpha_fast" value="0.0"/>
        <!-- <param name="laser_max_range" value="10.0"/>
        <param name="laser_min_range" value="0.08"/>-->

        <!--for MARKER PF-->
        <rosparam command="load" file="$(arg marker_positions_file)"/> <!-- load the marker positions -->
        <rosparam command="load" file="$(arg cameras_positions_file)"/> <!-- load the cameras positions -->
        <!--<param name="IMAGE_WIDTH" value= "$(arg width_image)" />
        <param name="IMAGE_HEIGHT" value= "$(arg height_image)" />-->
        <!--param name="MARKER_HEIGHT" value = "$(arg height_marker)" />
        <param name="MARKER_WIDTH" value = "$(arg width_marker)" / -->
        <param name="NUM_CAM" value ="1" /> <!-- Only work with 1 camera, do not change it-->
        <param name = "marker_model_type" value="observation_likelihood"/>

        <param name="use_map_topic" value="true"/>
        <param name="marker_z_hit" value="0.995"/>
        <param name="marker_z_rand" value ="0.005"/>
        <param name="marker_sigma_hit" value="100"/>
        <param name="marker_landa" value="2"/>
        <param name="base_frame_id" value="$(arg base_frame_robot)" /> 
        <param name="odom_frame_id" value="$(arg odom_topic)"/>
        <param name="global_frame_id" value="$(arg global_frame)"/>
        <param name="camera_topic" value="$(arg camera_topic)"/>
        <param name="height_pos_camera_link" value="$(arg height_pos_camera_link)"/>
        <param name="camera_link" value="$(arg camera_link)"/>
        <param name="topic_output_video" value="$(arg topic_output_video)"/>
        <param name="topic_marker" value="$(arg topic_marker)"/>

        <!--for FUSION laser-camera-->
        <param name="laser_coeff" value="1.0"/>
        <param name="camera_coeff" value="0.0"/>

        <!-- simulation or real (Only work with value 1 = simulation, do not change it) -->
        <param name="simulation" value = "1"/>
	</node>
</launch>
